app/services/users/profile.rb

```diff
@@ -63,6 +65,22 @@ def account
       nil
     end
 
+    # Get the preferred named from the user's demographics. See scaffold.errors for any errors.
+    #
+    # @return the user's preferred name or nil if the user has no preferred name or there was an
+    # error fetching the name.
+    #
+    def demographics_data
+      demographic_svc = VAProfile::Demographics::Service.new @user
+      demographics = demographic_svc.get_demographics.demographics
+      {
+        preferred_name: demographics.preferred_name.text
```

Comment:
Do you want to use safe method chaining here or are you just letting the rescue take care of any issues with obtaining demographic info?

Reply:
We wanted the rescue to take care of that, so we can catch any API issues like text is not supported, etc.